Makefile.work

```diff
@@ -112,9 +112,20 @@ endif
 rules/config.user:
 	@echo -n ""
 
+ifneq ($(DEFAULT_CONTAINER_REGISTRY),)
+SONIC_DEFAULT_CONTAINER_REGISTRY := $(DEFAULT_CONTAINER_REGISTRY)
+endif
+
 include rules/config
 -include rules/config.user
 
+ifeq ($(SONIC_DEFAULT_CONTAINER_REGISTRY),)
+SONIC_DEFAULT_CONTAINER_REGISTRY := $(DEFAULT_CONTAINER_REGISTRY)
+endif
+ifneq ($(SONIC_DEFAULT_CONTAINER_REGISTRY),)
+SONIC_DEFAULT_CONTAINER_REGISTRY := $(SONIC_DEFAULT_CONTAINER_REGISTRY)/
+endif
+
```

Comment:
Looks like the logic can be simplified, reduce one if block.

DEFAULT_CONTAINER_REGISTRY is from rule/config, SONIC_DEFAULT_CONTAINER_REGISTRY is the user input.
We can simply override the DEFAULT_CONTAINER_REGISTRY, when SONIC_DEFAULT_CONTAINER_REGISTRY is not empty.

ifneq ($(SONIC_DEFAULT_CONTAINER_REGISTRY),)
DEFAULT_CONTAINER_REGISTRY := $(SONIC_DEFAULT_CONTAINER_REGISTRY)
endif

When passing the value to j2, we can define a new parameter, for example:

ifneq ($(DEFAULT_CONTAINER_REGISTRY),)
DEFAULT_CONTAINER_REGISTRY_PREFIX:= $(DEFAULT_CONTAINER_REGISTRY)/
endif